From cde09b547abbeb7595d91259fcc628504d0f55a9 Mon Sep 17 00:00:00 2001
From: Marc Fournier <marc.fournier@camptocamp.com>
Date: Wed, 8 Jan 2014 18:05:40 +0100
Subject: [PATCH 05/22] Revert "curl_xml.c: avoid using uninitalized variable
 in error message"

This reverts commit bbbf37d55a3959951604c4be482e9a705a0f86d9.

A different fix for this problem was commited to the collectd-5.4 branch
(8327ee64) and conflicts with this patch. Also, initializing url the
same way as 8327ee64 does it, to avoid a merge conflict.
---
 src/curl_xml.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/curl_xml.c b/src/curl_xml.c
index 77aee60..b941f02 100644
--- a/src/curl_xml.c
+++ b/src/curl_xml.c
@@ -550,13 +550,14 @@ static int cx_curl_perform (cx_t *db, CURL *curl) /* {{{ */
   long rc;
   char *ptr;
   char *url;
+  url = db->url;
 
-  db->buffer_fill = 0;
+  db->buffer_fill = 0; 
   status = curl_easy_perform (curl);
   if (status != CURLE_OK)
   {
-    ERROR ("curl_xml plugin: curl_easy_perform failed with status %i: %s",
-           status, db->curl_errbuf);
+    ERROR ("curl_xml plugin: curl_easy_perform failed with status %i: %s (%s)",
+           status, db->curl_errbuf, url);
     return (-1);
   }
 
-- 
1.9.3

